File: tests/fixtures/sample.scala
package com.example

case class Point(x: Int, y: Int)

trait Renderable {
  def render(): String
}

class Calculator(private var value: Int = 0) {
  def add(x: Int): Int = {
    value += x
    value
  }

  private def reset(): Unit = {
    value = 0
  }
}

object Logger {
  def log(msg: String): Unit = {
    println(msg)
  }
}

enum Color {
  case Red, Green, Blue
}

def greet(name: String): String = {
  s"Hello, $name!"
}

def add(a: Int, b: Int): Int = a + b

File: tests/fixtures/sample.c
#include <stdio.h>

struct Point {
    int x;
    int y;
};

enum Color {
    RED,
    GREEN,
    BLUE
};

typedef struct {
    int width;
    int height;
} Rectangle;

int add(int a, int b) {
    return a + b;
}

void print_point(struct Point p) {
    printf("(%d, %d)\n", p.x, p.y);
}

static int helper(int x) {
    return x * 2;
}
